csv_detective/utils.py

```diff
@@ -58,19 +59,26 @@ def test_col(table, all_tests, num_rows, output_mode):
     test_funcs = dict()
     for test in all_tests:
         name = test.__name__.split('.')[-1]
-
         test_funcs[name] = {
             'func': test._is,
             'prop': test.PROPORTION
         }
-
     return_table = pd.DataFrame(columns=table.columns)
     for key, value in test_funcs.items():
+        # When analysis of all file is requested (num_rows = -1) we fix a threshold of 
+        # 1000 rows for every checks outside int or float format
+        local_num_rows = max(num_rows, 1000)
+        # For checks detecting int or float format, we analyze the whole file (because
+        # error can be generated afterward when exploiting this data into a database)
+        if key in ["int", "float", "longitude", "longitude_l93", "longitude_wgs", 
+            "longitude_wgs_fr_metropole", "latitude", "latitude_l93", "latitude_wgs", "latitude_wgs_fr_metropole", "iso_country_code_numeric"
+        ]:
+            local_num_rows = max(-1, num_rows)
```

Comment:
How about (for lines 70 to 76):
```
if num_rows==-1:

    if key in ["int", "float", "longitude", "longitude_l93", "longitude_wgs", 
            "longitude_wgs_fr_metropole", "latitude", "latitude_l93", "latitude_wgs", "latitude_wgs_fr_metropole", "iso_country_code_numeric"
        ]:

        local_num_rows=-1

    else:

        local_num_rows=1000

else:

    local_num_rows=min(rows,1000)
```

Reply:
With this : 
- If we setup an analyze to 500 rows, we will analyze more than user asked
- If we setup an analyze to 2000 rows, we will analyze less than user asked